common/src/main/java/tachyon/util/io/BufferUtils.java

```diff
@@ -231,7 +231,7 @@ public static boolean equalIncreasingByteBuffer(int start, int len, ByteBuffer b
       return false;
     }
     buf.rewind();
-    if (buf.remaining() != len) {
+    if (buf.remaining() < len) {
```

Comment:
I made the change to using a strict equals. I looked at uses of the equalIncreasingByteArray and they all look good. More importantly, "mvn test" passed.